lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,235 @@
+var edx = edx || {},
+
+    Dropdown = (function() {
+
+        var _dropdown = {
+
+            opts: {
+                page: $(document),
+                midpoint: ($(window).width() / 2),
+                button: $('.button-more.has-dropdown'),
+                button_active: 'is-active',
+                menu: $('.dropdown-menu'),
+                menu_inactive: 'is-hidden',
+                menu_active: 'is-visible',
+                menu_align: 'align-',
+                text_align: 'control-text-align'
+            },
+
+            init: function(parent) {
+                var that = _dropdown;
+
+                if (that.opts.button.length && that.opts.menu.length) {
+
+                    if (parent) {
+                        that.opts.page = $(parent);
+                    }
+
+                    _dropdown.listenForClick();
+                    _dropdown.listenForKeypress();
+                }
+            },
+
+            listenForClick: function() {
+                var that = _dropdown;
+
+                that.opts.button.on('click', function() {
+                    that.closeDropdownMenu();
+                    that.openDropdownMenu($(this));
```

Comment:
I'm not clear why this is calling openDropdownMenu after closeDropdownMenu. Can you add a comment?